src/libraries/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -176,6 +203,21 @@ private async ValueTask<HttpResponseMessage> SendAsyncCore(HttpRequestMessage re
                 {
                     activity.SetEndTime(DateTime.UtcNow);
 
+                    if (activity.IsAllDataRequested)
+                    {
+                        // Set tags known at activity Stop.
+                        if (response is not null)
+                        {
```

Comment:
@antonfirsov - We also need to set the status of activity appropriately here.
ref :[Spec](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#status:~:text=For%20HTTP%20status%20codes%20in%20the%204xx%20range%20span%20status%20MUST%20be%20left%20unset%20in%20case%20of%20SpanKind.SERVER%20and%20MUST%20be%20set%20to%20Error%20in%20case%20of%20SpanKind.CLIENT)

Here is the implementation from instrumentation library: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/92f4b83755189a5d104a3535c9decbae5f4681e6/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpHandlerDiagnosticListener.cs#L173-L196